cpp/src/arrow/stl.h

```diff
@@ -441,6 +445,126 @@ Status TupleRangeFromTable(const Table& table, const compute::CastOptions& cast_
 }
 
 }  // namespace stl
```

Comment:
Now we have stuff inside and outside the `arrow::stl` namespace. Perhaps we could put everything inside?

Reply:
Agreed, and maybe rename `stl_allocator` to just `allocator`

Reply:
Thanks for the comments, would this logic apply to stl_test.cc as well.  As in put everything into stl namespace?

Reply:
Yes

Reply:
Yes